lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -502,6 +502,13 @@ pub trait SegmentOptimizer {
 
         let mut optimized_segment: Segment = segment_builder.build(indexing_permit, stopped)?;
 
+        // Delete points
+        let deleted_points_snapshot = proxy_deleted_points
+            .read()
+            .iter()
+            .map(|(point_id, versions)| (*point_id, *versions))
+            .collect::<Vec<_>>();
+
```

Comment:
What is the reason for moving this up? It does not have any affect on business logic does it?

Reply:
I had suspicions I could not confirm. I propose to keep it as in the worst case it just doesn't change anything